biothings/hub/databuild/auto_archive.py

```diff
@@ -0,0 +1,90 @@
+import datetime
+from functools import partial
+
+from dateutil import parser as dtparser
+
+from biothings import config as btconfig
+from biothings.utils.manager import BaseManager
+
+logger = btconfig.logger
+
+
+class AutoArchiveManager(BaseManager):
+    """This feature will add a new console command "auto_archive", a new api "/auto_archive".
+    It is intended to allow automatically archive old build, based on configuration.
+
+    This feature can be configurated by using AUTO_ARCHIVE_CONFIG variable, to determine:
+        - schedule: how frequency a build configuration should run
+        - days: how old a build should be archived
+
+    AUTO_ARCHIVE_CONFIG = {
+        "build_configuration_name": {
+            "schedule: "* 0 * * *",  # run daily at 0am UTC
+            "days": 3
+        },
+        ...
+    }
+    """
+
+    DEFAULT_ARCHIVE_AFTER_DAYS = 3
+    DEFAULT_ARCHIVE_SCHEDULE = "* 0 * * *"  # run daily at 0am UTC
```

Comment:
@sengineer0 when `AUTO_ARCHIVE_CONFIG` is not set, or the build name is not set in `AUTO_ARCHIVE_CONFIG`, the default behavior should be not doing auto_archive at all.